pallets/phala-world/src/traits.rs

```diff
@@ -113,3 +113,58 @@ pub struct FoodInfo<BoundedOriginOfShellsFed> {
 	/// A BoundedVec of (CollectionId, NftId)
 	pub origin_of_shells_fed: BoundedOriginOfShellsFed,
 }
+
+/// Shell Part types to determine if a ShellPartInfo is BasicPart, ComposablePart or a SubPart
+#[derive(Encode, Decode, Debug, Clone, Copy, Eq, PartialEq, TypeInfo, MaxEncodedLen)]
+#[cfg_attr(feature = "std", derive(Serialize, Deserialize))]
+pub enum ShellPartType {
+	BasicPart,
+	ComposablePart,
+	SubPart,
+}
+
+/// Shell Parts info
+#[derive(Encode, Decode, Eq, PartialEq, Clone, RuntimeDebug, TypeInfo, MaxEncodedLen)]
+#[cfg_attr(feature = "std", derive(Serialize, Deserialize))]
+pub struct ShellPartInfo<BoundedString, BoundedSubParts> {
+	/// Name of the Part
```

Comment:
Looks like `ShellPartInfo` always has the same fields as `ShellSubPartInfo`. You can write in this way:

```
pub struct ShellPartInfo {
  shell: PartInfo,
  sub_parts: BoundedSubParts,
}
pub struct PartInfo {
  // ...
}
```

Reply:
Done